### PR TITLE
make profiling work

### PR DESCRIPTION
**Pull request by fffw**

### CODE REVIEW

File: src/github.com/getlantern/flashlight/flashlight.go [hunk 1]
```diff
@@ -170,8 +149,6 @@ func doMain() error {
 	}
 	displayVersion()
 
-	parseFlags()
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/flashlight.go]: duplicate with L127

File: src/github.com/getlantern/flashlight/flashlight.go [hunk 2]
```diff
@@ -194,8 +193,11 @@ func doMain() error {
 			exit(fmt.Errorf("Wrong arguments"))
 		}
 
-		finishProfiling := profiling.Start(cfg.CpuProfile, cfg.MemProfile)
-		defer finishProfiling()
+		if cfg.CpuProfile != "" || cfg.MemProfile != "" {
+			log.Debugf("Start profiling with cpu file %s and mem file %s", cfg.CpuProfile, cfg.MemProfile)
+			finishProfiling := profiling.Start(cfg.CpuProfile, cfg.MemProfile)
+			addExitFunc(finishProfiling)
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/flashlight.go]: `defer` will never execute as we are using `exit()`

File: src/github.com/getlantern/flashlight/flashlight.go [hunk 3]
```diff
@@ -205,9 +207,9 @@ func doMain() error {
 		log.Debug("Running proxy")
 		if cfg.IsDownstream() {
 			// This will open a proxy on the address and port given by -addr
-			go runClientProxy(cfg)
+			runClientProxy(cfg)
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/flashlight.go]: No need to spawn goroutine here as the code block is already wrapped in a goroutine.

**Comment by atavism** [comment]
LGTM!

**Pull request merged**